pkg/discovery/service.go

```diff
@@ -158,6 +159,24 @@ func (discovery *Discovery) refresh(ctx context.Context) error {
 		if err != nil {
 			discovery.log.Error("could not put node into cache", zap.String("ID", ping.Id.String()), zap.Error(err))
 		}
+
+		// update email and wallet with correct info
+		_, info, err := discovery.kad.FetchInfo(ctx, &pb.NodeAddress{
+			Address: ping.GetAddress().String(),
+		})
+		if err != nil {
+			discovery.log.Warn("could not fetch node info", zap.String("ID", ping.GetAddress().String()))
+		}
+
+		_, err = discovery.statdb.UpdateOperator(ctx, ping.Id, &statdb.NodeStats{
```

Comment:
This call should happen only if `discovery.kad.FetchInfo` above was successful.

Reply:
Fixed